### PR TITLE
Document setting user filter on spring for transactions

### LINKED ISSUE #3482: Document setting user filter on spring for transactions
## Issue

Document setting user filter on spring for transactions

## Description

A new filter should be added to get users set into transactions
related to https://github.com/getsentry/sentry-java/pull/1443

### Suggested Solution

Add user filter in performance setup and add an entry to the migration page

### PR DESCRIPTION
**Pull request by maciejwalkowiak**
Fixes #3482

### CODE REVIEW

File: src/platforms/java/guides/spring/advanced-usage.mdx [hunk 1]
```diff
@@ -29,6 +31,133 @@ import io.sentry.spring.EnableSentry
 class SentryConfiguration
 ```
 
+2. Register servlet filter bean `SentryUserFilter`:
```
Comment (PeloWriter) [inline src/platforms/java/guides/spring/advanced-usage.mdx:34]:
```suggestion
2. Register the servlet filter bean `SentryUserFilter`:
```

File: src/platforms/java/guides/spring/advanced-usage.mdx [hunk 2]
```diff
@@ -29,6 +31,133 @@ import io.sentry.spring.EnableSentry
...
+        val sentryUserFilterProxy = DelegatingFilterProxy()
+        sentryUserFilterProxy.setTargetBeanName("sentryUserFilter")
+        return arrayOf(
+            springSecurityFilterChain, requestContextFilter, sentryUserFilterProxy
+        )
+    }
+}
+```
+
+By default, the username is retrieved from `HttpServletRequest#userPrincipal`. To retrieve username from Spring Security context, register `SpringSecuritySentryUserProvider` bean:
```
Comment (PeloWriter) [inline src/platforms/java/guides/spring/advanced-usage.mdx:131]:
```suggestion
By default, the username is retrieved from `HttpServletRequest#userPrincipal`. To retrieve the username from Spring Security context, register the `SpringSecuritySentryUserProvider` bean:
```

**Review by PeloWriter — APPROVED** [review]

**Comment by maciejwalkowiak** [comment]
Thank you @PeloWriter for polishing!

File: src/platforms/java/guides/spring/advanced-usage.mdx [hunk 3]
```diff
@@ -8,7 +8,9 @@ Record user information from an HTTP request or by registering a Spring bean for
 
 ## Recording User Information From HTTP Request
 
-To record the user's IP address and `Principal#name` as the username, set the personal information flag to `true`.
+To record the user's IP address and `Principal#name` as the username:
```
Comment (marandaneto) [inline src/platforms/java/guides/spring/advanced-usage.mdx:11]: should we link this section to the performance page?

**Comment by marandaneto** [comment]
making it a draft till we publish it

File: src/platforms/java/guides/spring/advanced-usage.mdx (hunk 3 above)
Comment (PeloWriter) [reply src/platforms/java/guides/spring/advanced-usage.mdx:11]: Great idea. I'd suggest:
```suggestion
To record the user's IP address and `Principal#name` as the username so you can then view in [Event Details](/product/performance/event-detail/:
```

File: src/platforms/java/guides/spring/advanced-usage.mdx [hunk 4]
```diff
@@ -29,6 +31,133 @@ import io.sentry.spring.EnableSentry
 class SentryConfiguration
 ```
 
+2. Register the servlet filter bean `SentryUserFilter`:
```
Comment (marandaneto) [inline src/platforms/java/guides/spring/advanced-usage.mdx:34]: this section could have a `note` which makes it available under 5.0 as a beta version still

File: src/platforms/java/guides/spring/advanced-usage.mdx [hunk 5]
```diff
@@ -29,6 +31,133 @@ import io.sentry.spring.EnableSentry
...
+      SentryOptions sentryOptions) {
+    return new SpringSecuritySentryUserProvider(sentryOptions);
+  }
+}
+```
+
+```kotlin
+import io.sentry.SentryOptions
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
```
Comment (marandaneto) [inline src/platforms/java/guides/spring/advanced-usage.mdx:152]: missing SpringSecuritySentryUserProvider

File: src/platforms/java/guides/spring/advanced-usage.mdx (hunk 4 above)
Comment (maciejwalkowiak) [reply src/platforms/java/guides/spring/advanced-usage.mdx:34]: Not sure how to phrase it specifically to this point. Feel free to create a PR :-)

Comment (marandaneto) [reply src/platforms/java/guides/spring/advanced-usage.mdx:34]: https://docs.sentry.io/contributing/pages/components/#alert

just add an alert before that explaining this is available under SDK v5 and its beta still, we've done that before when SDKs were beta or when features were beta too, pretty sure you can make it :)

File: src/platforms/java/guides/spring/advanced-usage.mdx [hunk 6]
```diff
@@ -8,7 +8,9 @@ Record user information from an HTTP request or by registering a Spring bean for
 
 ## Recording User Information From HTTP Request
 
-To record the user's IP address and `Principal#name` as the username, set the personal information flag to `true`.
+To record the user's IP address and `Principal#name` as the username so you can then view in [Event Details](/product/performance/event-detail/:
```
Comment (marandaneto) [inline src/platforms/java/guides/spring/advanced-usage.mdx:11]: @maciejwalkowiak link is broken, you can see the preview via vercel https://sentry-docs-git-fork-maciejwalkowiak-gh-3482.sentry.dev/platforms/java/guides/spring/advanced-usage/

Comment (marandaneto) [inline src/platforms/java/guides/spring/advanced-usage.mdx:11]:
```suggestion
To record the user's IP address and `Principal#name` as the username so you can then view in [Event Details](/product/performance/event-detail/):
```

**Pull request merged**